fix: Filter unsupported JSON Schema attributes out of Gemini tool calls

necaris opened the pull request:
As noted in various places, including [LangChain](https://github.com/langchain-ai/langchainjs/pull/5247), Gemini's support for JSON Schema is incomplete. The `additionalProperties` field and the `format` property on `string` types are not supported.

This filters them out of Gemini API tool calls, so that tools that define these properties (e.g. from MCP servers) don't immediately cause a 400 when invoking Gemini.

karthink commented on gptel-gemini.el:
```diff
@@ -143,6 +143,25 @@ list."
      (gptel-backend-request-params gptel-backend)
      (gptel--model-request-params  gptel-model))))
 
+(defvar gptel--gemini-unsupported-json-attributes '(:additionalProperties)
```
On second thought, we can remove this variable entirely, and just hard code `:additionalProperties` into the filtering function.  In a code comment in the function you can mention that only `:additionalProperties` is problematic right now.

If the list of unsupported attribute grows we can introduce a `defconst`.

necaris commented:
Thank you @karthink for the quick & thorough review! Addresses all your comments, and went ahead and reworded the commit to fit the pattern of previous commits too.

karthink commented:
@necaris Looks great, very clean!  Thank you for the PR.
